creating_geometric_models/create_fault_from_trace.py

```diff
@@ -63,26 +63,85 @@ def compute_rel_curvilinear_coordinate(nodes):
     reldist[reldist > 1] = 1.0
     return reldist
 
+def generate_vertices_constant_dip(maxdepth, sign=1):
+    """
+    extends 2d fault trace (nodes array) along constant dip
+    returns a 3d array of vertices coordinates
+    """
+    nd = int(maxdepth / (sin(dip) * dx))
+    vertices = np.zeros((nx, nd, 3))
+    vertices[:, 0, :] = nodes
+    one_over_tan_dip = 1.0 / tan(dip)
+    for i in range(0, nx):
+        ud = -(one_over_tan_dip * av1[i, :] - uz)
+        ud = ud / np.linalg.norm(ud)
+        for j in range(1, nd):
+            vertices[i, j, :] = vertices[i, j - 1, :] - sign * dx * ud
+    return vertices
+
+
+def generate_vertices_dip_vary_along_strike(maxdepth, sign=1):
+    """
+    extends 2d fault trace (nodes array) along dip, with dip varying along strike
+    In this case we use constant depth strides between along dip vertices
+    that is the size of the mesh along dip will vary with dip
+    returns a 3d array of vertices coordinates
+    """
+    nd = int(maxdepth / dx)
+    vertices = np.zeros((nx, nd, 3))
+    vertices[:, 0, :] = nodes
+    for i in range(0, nx):
+        one_over_tan_dip = 1.0 / tan(aDip[i])
+        for j in range(1, nd):
+            ud = -(one_over_tan_dip * av1[i, :] - uz)
+            vertices[i, j, :] = vertices[i, j - 1, :] - sign * dx * ud
+    return vertices
+
 
-def generate_vertices(depth, sign=1):
+def generate_vertices_dip_vary_along_dip(maxdepth, sign=1):
+    """
+    extends 2d fault trace (nodes array) along depth-dependent dip
+    returns a 3d array of vertices coordinates
+    """
+    # compute depth array
+    current_depth = 0.0
+    depth = []
+    while True:
```

Comment:
Couldn't this be while current_depth < maxdepth?

Reply:
yes <=